module/VuFind/src/VuFind/IdentifierLinker/BrowZine.php

```diff
@@ -0,0 +1,197 @@
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:record_drivers Wiki
+ */
+
+namespace VuFind\IdentifierLinker;
+
+use VuFind\I18n\Translator\TranslatorAwareInterface;
+use VuFindSearch\Backend\BrowZine\Command\LookupDoiCommand;
+use VuFindSearch\Backend\BrowZine\Command\LookupIssnsCommand;
+use VuFindSearch\Service;
+
+use function in_array;
+
+/**
+ * BrowZine DOI linker
+ *
+ * @category VuFind
+ * @package  DOI
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:record_drivers Wiki
+ */
+class BrowZine implements IdentifierLinkerInterface, TranslatorAwareInterface
+{
+    use \VuFind\I18n\Translator\TranslatorAwareTrait;
+
+    /**
+     * Constructor
+     *
+     * @param Service $searchService Search service
+     * @param array   $config        Configuration settings
+     * @param array   $doiServices   Configured DOI services
+     * @param array   $issnServices  Configured ISSN services
+     */
+    public function __construct(
+        protected Service $searchService,
+        protected array $config = [],
+        protected array $doiServices = [],
+        protected array $issnServices = []
+    ) {
+    }
+
+    /**
+     * Check if an array key is available in the data and allowed by filter settings.
+     *
+     * @param string $key  Key to check
+     * @param array  $data Available data
+     *
+     * @return bool
+     */
+    protected function arrayKeyAvailable(string $key, ?array $data): bool
+    {
+        if (empty($data[$key])) {
+            return false;
+        }
+        switch (strtolower(trim($this->config['filterType'] ?? 'none'))) {
+            case 'include':
+                return in_array($key, (array)($this->config['filter'] ?? []));
+            case 'exclude':
+                return !in_array($key, (array)($this->config['filter'] ?? []));
+            default:
+        }
+        // If we got this far, no filter setting is applied, so the option is legal:
+        return true;
+    }
+
+    /**
+     * Format a single service link.
+     *
+     * @param array  $data       Raw API response data
+     * @param string $serviceKey Key being extracted from response
+     * @param array  $config     Service-specific configuration settings
+     *
+     * @return array{link: string, label: string, data: array, localIcon: ?string, icon: ?string}
+     */
+    protected function processServiceLink(array $data, string $serviceKey, array $config): array
+    {
+        $result = [
+            'link' => $data[$serviceKey],
+            'label' => $this->translate($config['linkText']),
+            'data' => $data,
+        ];
+        $localIcons = !empty($this->config['local_icons']);
+        if (!$localIcons && !empty($config['icon'])) {
```

Comment:
Small suggestion to remove double negation:

```suggestion
        $localIcons = $this->config['local_icons'] ?? false;
        if ($localIcons && !empty($config['icon'])) {
```

Reply:
We only want to use non-local icons when the local_icons setting is false -- so I don't think changing the negation of that setting is correct. With this change, we would show non-local icons when local_icons is set to true.

Reply:
Yes of course, you are right.